op-node/node/node.go

```diff
@@ -42,6 +42,8 @@ type closableSafeDB interface {
 }
 
 type OpNode struct {
+	// Retain the config to test for active features rather than test for runtime state.
+	cfg        *Config
```

Comment:
Why is this actually better? It seems more robust to me to check at runtime whether a field is nil instead of assuming that the `OpNode` has been created with the correct constructor and configuration? Couldn't there be scenarios in e.g. testing where an OpNode struct is directly created and then it's safer to just test directly e.g. `n.p2pNode != nil`?

So in that sense maybe the node `Config` is something that is consumed by the constructor but not necessarily valid or required after.

Reply:
The reason is that checking for nil is defensive programming. We have a clear predicate for when p2p is enabled so we should use that. To phrase it another way, nil-checking/defensive programming creates additional program states that we can't and don't account for. p2pNode != nil is being used here for testing if it's been shutdown but also for whether p2p is expected to exist and that partial state propagates to other areas of the code that are trying to call in with events.

Reply:
As it happens, it's always created with the config. Per the other comment about weird states, I think it's worth keeping it in.